### PR TITLE
Add new links to experimental docs and update JS testing docs

### PR DESCRIPTION
**Pull request by opr**
As part of my porter duties I reviewed our current [documentation for feature flags and experimental interfaces](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/blocks/feature-flags-and-experimental-interfaces.md) and noticed there were some out of date/no longer used occurrences.

Specifically the largest change was removing links to out of date references in the `Product Price new controls ` section. The file `assets/js/atomic/blocks/product-elements/price/block.js` had been changed to use `ProductPrice` in #3065 so some of the `isFeaturePluginBuild()` calls were no longer used and this document was stale.

I added a line to the documentation about `__experimental_woocommerce_blocks_payment_gateway_features_list ` which was added in #3719 but not documented. 

This PR will bring that document in line with the current state of the project and provide a more accurate documentation of our use of experimental interfaces/feature flags.

I also made a change to our current [JS testing documentation](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/contributors/javascript-testing.md). Because it referenced Travis, and I also changed some version numbers in there to make it a little more up to date.

### How to test the changes in this Pull Request:

1. View the **new** [documentation for feature flags and experimental interfaces](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/d4f32877d8697752d621aff9b30915a3432d8c9a/docs/blocks/feature-flags-and-experimental-interfaces.md) 👀 and verify that it's correct.
2. View the **new** [documentation for JS testing](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/d4f32877d8697752d621aff9b30915a3432d8c9a/docs/contributors/javascript-testing.md) 👀 and verify it's correct.

### Changelog

> Improvements to documentation.

### CODE REVIEW

File: docs/blocks/feature-flags-and-experimental-interfaces.md [hunk 1]
```diff
@@ -45,13 +45,14 @@ We also have individual features or code blocks behind a feature flag, this is a
 ## Processes and commands that use a flag
 
 -   `npm run build:deploy` uses the feature plugin flag ([env flag](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/c0de18ec0a798c072420c67a689e4cc4d3ac77c9/package.json#L28)).
--   travis uses the experimental flag when running tests ([env flags](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/d6a506c19056d42d40743ae0126153fb81004672/.travis.yml#L68-L143)).
+-   GitHub actions uses the experimental flag when running automated tests ([env flags](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/.github/workflows/php-js-e2e-tests.yml)).
```
Comment (ralucaStan) [inline docs/blocks/feature-flags-and-experimental-interfaces.md:48]: Would it help if we added a code line to the link? Not sure if it's the right line tho` `https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/.github/workflows/php-js-e2e-tests.yml#L263`

**Review by ralucaStan — APPROVED** [review]
Really useful update. 👍 Added a small comment, but is not essential, so I am approving.

File: docs/blocks/feature-flags-and-experimental-interfaces.md (hunk 1 above)
Comment (opr) [reply docs/blocks/feature-flags-and-experimental-interfaces.md:48]: That flag is used in several places in that file, so I didn't want to just link to one instance, for fear of implying that it was an isolated use. We could link to each use actually, and very explicitly document each individual use. What do you think?

Comment (ralucaStan) [reply docs/blocks/feature-flags-and-experimental-interfaces.md:48]: I think that would be hard to maintain. It's ok to leave it as it is.

**Pull request merged**